sources/dogtag/bin/imds.rs

```diff
@@ -0,0 +1,44 @@
+/*!
+dogtag is a set of tools that detect the hostname of a bottlerocket server/instance and prints it to stdout.
+if the tool is called in an environment it cannot resolve the hostname it will error out.
+
+Currently the following hostname tools are implemented:
+
+* 01-imds - Fetches hostname from the Instance Metadata via IMDS
+* 00-reverse-dns - Uses reverse dns lookup to resolve the hostname
+ */
+use dogtag::Cli;
+use imdsclient::ImdsClient;
+use snafu::{OptionExt, ResultExt};
+
+type Result<T> = std::result::Result<T, error::Error>;
+
+/// Implements a hostname lookup tool by fetching the public hostname
+/// from the instance metadata via IMDS. It will interface with IMDS
+/// via:
+///
+/// * Check for IPv6, default to IPv4 if not available
+/// * Check for IMDSv2, fallback to IMDSv1 if not enabled
+#[tokio::main]
+async fn main() -> Result<()> {
+    // Even though for this helper we do not need any arguments
+    // still validate to ensure the helper follows standards.
+    let _: Cli = argh::from_env();
+    let mut imds = ImdsClient::new();
+    let hostname = imds.fetch_hostname().await.context(error::ImdsSnafu)?;
+    println!("{}", hostname.context(error::NoHostnameSnafu)?);
```

Comment:
nit: I think I'd prefer we handle all the errors right away, e.g.:

```suggestion
    let hostname = imds.fetch_hostname().await
            .context(error::ImdsSnafu)?
            .context(error::NoHostnameSnafu)?;
    println!("{}", hostname);
```